voxblox_ros/src/tsdf_server.cc

```diff
@@ -193,13 +134,126 @@ TsdfServer::TsdfServer(const ros::NodeHandle& nh,
+    const ros::NodeHandle& nh_private) {
+  // Before subscribing, determine minimum time between messages.
+  // 0 by default.
+  double min_time_between_msgs_sec = 0.0;
+  nh_private.param("min_time_between_msgs_sec", min_time_between_msgs_sec,
+                   min_time_between_msgs_sec);
+  min_time_between_msgs_.fromSec(min_time_between_msgs_sec);
+
+  nh_private.param("max_block_distance_from_body",
+                   max_block_distance_from_body_,
+                   max_block_distance_from_body_);
+  nh_private.param("slice_level", slice_level_, slice_level_);
+  nh_private.param("world_frame", world_frame_, world_frame_);
+  nh_private.param("publish_pointclouds_on_update",
+                   publish_pointclouds_on_update_,
+                   publish_pointclouds_on_update_);
+  nh_private.param("publish_slices", publish_slices_, publish_slices_);
+  nh_private.param("publish_pointclouds", publish_pointclouds_,
+                   publish_pointclouds_);
+
+  nh_private.param("use_freespace_pointcloud", use_freespace_pointcloud_,
+                   use_freespace_pointcloud_);
+  nh_private.param("pointcloud_queue_size", pointcloud_queue_size_,
+                   pointcloud_queue_size_);
+  nh_private.param("enable_icp", enable_icp_, enable_icp_);
+  nh_private.param("accumulate_icp_corrections", accumulate_icp_corrections_,
+                   accumulate_icp_corrections_);
+
+  nh_private.param("verbose", verbose_, verbose_);
+
+  // Mesh settings.
+  nh_private.param("mesh_filename", mesh_filename_, mesh_filename_);
+  std::string color_mode("");
+  nh_private.param("color_mode", color_mode, color_mode);
+  color_mode_ = getColorModeFromString(color_mode);
+
+  // Color map for intensity pointclouds.
+  std::string intensity_colormap("rainbow");
+  float intensity_max_value = kDefaultMaxIntensity;
+  nh_private.param("intensity_colormap", intensity_colormap,
+                   intensity_colormap);
+  nh_private.param("intensity_max_value", intensity_max_value,
+                   intensity_max_value);
+
+  // Default set in constructor.
+  if (intensity_colormap == "rainbow") {
+    color_map_.reset(new RainbowColorMap());
+  } else if (intensity_colormap == "inverse_rainbow") {
+    color_map_.reset(new InverseRainbowColorMap());
+  } else if (intensity_colormap == "grayscale") {
+    color_map_.reset(new GrayscaleColorMap());
+  } else if (intensity_colormap == "inverse_grayscale") {
+    color_map_.reset(new InverseGrayscaleColorMap());
+  } else if (intensity_colormap == "ironbow") {
+    color_map_.reset(new IronbowColorMap());
+  } else {
+    ROS_ERROR_STREAM("Invalid color map: " << intensity_colormap);
+  }
+  color_map_->setMaxValue(intensity_max_value);
 }
 
 // Check if all coordinates in the point are finite
-template<typename Point>
-bool isPointFinite(const Point& point) {
-    return std::isfinite(point.x) && std::isfinite(point.y) &&
-           std::isfinite(point.z);
+template <typename PCLPoint>
```

Comment:
do we have a tools/utils header somewhere we can stick this conversion stuff?

Reply:
moved to voxblox_ros/conversions.h